### ISSUE TITLE
Annoying taskbar icon

### ISSUE THREAD

**Issue by takase1121** [rocksdanister/lively#78]
**Describe the bug**
I think Windows update probably broke something. Lively wallpaper now displays a taskbar entry when using video as wallpaper. I just installed this, I don't know how it looks like before the update so I might be wrong

**To Reproduce**
1. Set a wallpaper
2. A taskbar entry  is shown

**Expected behavior**
There shouldn't be a taskbar entry.

**Screenshots/Video**
If applicable, add screenshots/video to help explain your problem.
With mpv as backend:
![image](https://user-images.githubusercontent.com/20792268/85051733-76087f80-b1ca-11ea-928c-5182c54c0dea.png)
With others it just displays "MediaKit" for LAV and "MediaPlayer" for MFC

**Desktop (please complete the following information):**
 - OS: Windows 10 2004

**Comment by rocksdanister** [comment]
While I did not try 2004 yet, others have it working fine: https://www.reddit.com/r/LivelyWallpaper/comments/gudq0m/has_anyone_tried_lively_with_windows_10_version/

Usually when you make a window a child of something (desktop in this case) - the taskbar Icon should disappear but in addition to that I'm changing the window style to force remove it from taskbar:

        public static void RemoveWindowFromTaskbar(IntPtr handle)
        {
            var styleNewWindowExtended =
                   (Int64)NativeMethods.WindowStyles.WS_EX_NOACTIVATE
                   | (Int64)NativeMethods.WindowStyles.WS_EX_TOOLWINDOW;

            // update window styles
            NativeMethods.SetWindowLongPtr(new HandleRef(null, handle), (-20), (IntPtr)styleNewWindowExtended);
        }

In my system this worked fine, I did notice it not doing anything in rare cases for some systems; not sure why.

Since you are using a webm file, as a temporary fix you can use the included webbrowser to play the videos:
https://github.com/rocksdanister/lively/wiki/Web-Guide-VI-:-Webpage-Video-Support

Or you can just use this wallpaper and set just the center panel to hidden:
https://www.reddit.com/r/LivelyWallpaper/comments/grxsth/simple_weather_video_wallpaper_customisable/

**Comment by takase1121** [comment]
TL;DR:
Change `Multitasking > Virtual Desktops > On the taskbar, show windows that are open on` to `Only the desktop I'm using`.


The web browser method has quite some caveats in my case:
- need some JS to make screen not blink when the video loops, because apparently the browser needs to reload data or something.
- suddenly not working after exiting some game, probably Windows suspended it for more resources
- unable to use H.264 because of licensing and patents probably (I can use VP9, but the next caveat explains why)
- CEF not using hardware accelerated video playback (instead it uses FFMPEG, which takes 10-20% CPU and stops laptop from suspending)
- Kind of slow to start compared to a video player (duh, its Chromium)

But then I remembered I choose `All desktops` for `Multitasking > Virtual Desktops > on the taskbar, show windows that are open on`. Apparently changing that to `Only the desktop I'm using` does the trick. I suspect that there is no easy fix to this, because its mostly Windows' fault. But at least now the video player doesn't appear in the taskbar anymore.

**Comment by rocksdanister** [comment]
That is very weird when ms docs itself saying for [WS_EX_TOOLWINDOW
](https://docs.microsoft.com/en-us/windows/win32/winmsg/extended-window-styles)

> A tool window has a title bar that is shorter than a normal title bar, and the window title is drawn using a smaller font. A tool window does not appear in the taskbar or in the dialog that appears when the user presses ALT+TAB. 

I did in the past set the window property in wpf code itself instead of doing it like this, but that was crashing when the windows timeline was opened.

**Comment by takase1121** [comment]
~~just Microsoft things~~ That intrigued me as well, I suspect that people at MS implemented WS_EX_TOOLWINDOW with Virtual Desktops, eg: A virtual desktop called `<insert annoying GUID here>` to hide their taskbar entries but I could be entirely wrong though.

**Comment by rocksdanister** [comment]
Fixed: https://github.com/rocksdanister/lively/commit/16764c7585d13775ae2ecdbe98d89b3c2e319347
